drivers/ieee802154/ieee802154_mcxw.c

```diff
@@ -0,0 +1,1346 @@
+ * aTimeUs, when the timer overflows. Therefore the formula should be:
+ *
+ * if (aTimeUs <= mcxw_ctx.csl_sample_time)
+ *         cslPhaseUs = mcxw_ctx.csl_sample_time - aTimeUs;
+ * else
+ *         cslPhaseUs = MAX_TIMER_VALUE - aTimeUs + mcxw_ctx.csl_sample_time;
+ *
+ * For simplicity the formula below has been used.
+ */
+static uint16_t rf_compute_csl_phase(uint32_t aTimeUs)
+{
+	/* convert CSL Period in microseconds - it was given in 10 symbols */
+	uint32_t cslPeriodUs = mcxw_ctx.csl_period * 10 * IEEE802154_SYMBOL_TIME_US;
+	uint32_t cslPhaseUs =
+		(cslPeriodUs - (aTimeUs % cslPeriodUs) + (mcxw_ctx.csl_sample_time % cslPeriodUs)) %
+		cslPeriodUs;
+
+	return (uint16_t)(cslPhaseUs / (10 * IEEE802154_SYMBOL_TIME_US) + 1);
+}
+#endif /* CONFIG_IEEE802154_CSL_ENDPOINT */
+
+/*************************************************************************************************/
+static void rf_abort(void)
+{
+	macToPlmeMessage_t msg;
+
+	sunRxMode = RX_ON_IDLE_START;
+	msg.msgType = gPlmeSetReq_c;
+	msg.msgData.setReq.PibAttribute = gPhyPibRxOnWhenIdle;
+	msg.msgData.setReq.PibAttributeValue = (uint64_t)0;
+
+	(void)MAC_PLME_SapHandler(&msg, ot_phy_ctx);
+
+	msg.msgType = gPlmeSetTRxStateReq_c;
+	msg.msgData.setTRxStateReq.state = gPhyForceTRxOff_c;
+
+	(void)MAC_PLME_SapHandler(&msg, ot_phy_ctx);
+}
+
+static void rf_set_channel(uint8_t channel)
+{
+	macToPlmeMessage_t msg;
+
+	msg.msgType = gPlmeSetReq_c;
+	msg.msgData.setReq.PibAttribute = gPhyPibCurrentChannel_c;
+	msg.msgData.setReq.PibAttributeValue = (uint64_t)channel;
+
+	(void)MAC_PLME_SapHandler(&msg, ot_phy_ctx);
+}
+
+static int mcxw_cca(const struct device *dev)
+{
+	macToPlmeMessage_t msg;
+	phyStatus_t phy_status;
+
+	struct mcxw_context *mcxw_radio = dev->data;
+
+	msg.msgType = gPlmeCcaReq_c;
+	msg.msgData.ccaReq.ccaType = gPhyCCAMode1_c;
+	msg.msgData.ccaReq.contCcaMode = gPhyContCcaDisabled;
+
+	phy_status = MAC_PLME_SapHandler(&msg, ot_phy_ctx);
+
+	__ASSERT_NO_MSG(phy_status == gPhySuccess_c);
+
+	k_sem_take(&mcxw_radio->cca_wait, K_FOREVER);
+
+	return (mcxw_radio->tx_status == OT_ERROR_CHANNEL_ACCESS_FAILURE) ? -EBUSY : 0;
+}
+
+static int mcxw_set_channel(const struct device *dev, uint16_t channel)
+{
+	struct mcxw_context *mcxw_radio = dev->data;
+
+	LOG_DBG("%u", channel);
+
+	if (channel != mcxw_radio->channel) {
+
+		if (channel < 11 || channel > 26) {
+			return channel < 11 ? -ENOTSUP : -EINVAL;
+		}
+
+		mcxw_radio->channel = channel;
+	}
+
+	return 0;
+}
+
+static net_time_t mcxw_get_time(const struct device *dev)
+{
+	static uint64_t u64SwTimestampUs = 0;
+	static uint64_t u64HwTimestampUs = 0;
+
+	/* Get new 32bit HW timestamp */
+	uint32_t u32Ticks;
+	uint64_t u64HwTimestampUs_new;
+	uint64_t wrapped_val = 0;
+	uint64_t increment;
+
+	if (counter_get_value(mcxw_ctx.counter, &u32Ticks)) {
+		return -1;
+	}
+
+	u64HwTimestampUs_new = counter_ticks_to_us(mcxw_ctx.counter, u32Ticks);
+
+	OSA_InterruptDisable();
```

Comment:
Move it before counter_get_value() or remove it

Reply:
removed